posts/20241209full-package-reviews/index.qmd

```diff
@@ -0,0 +1,57 @@
+---
+title: "How we work: My first full package review"
+author:
+  - name: "Chris Hartgerink"
+    orcid: "0000-0003-1050-6809"
+date: "2024-12-09"
+categories: [blueprints, "How we work"]
+format:
+  html:
+    toc: true
+---
+
+At Epiverse-TRACE, we continuously develop packages, which risks making us (hyper)focused on the everyday changes. Feature here, bug fix there. We can call this "development by Pull Request," where we introduce changes in small batches. Of course, it is a developer's responsibility to also take into consideration the entire codebase. However, to sometimes enforce taking that step back, [our contribution blueprints outline that we do full package reviews](https://epiverse-trace.github.io/blueprints/code-review.html#full-package-review). In this blog post, I highlight what a full package review is, how it is set up, and some of my experiences with this process.
```

Comment:
Does 'development by PR' necessarily equate small batch changes? Could be useful to point out the common alternative?

Reply:
Thanks for the comment 😊 You are right, it is not necessarily small. I think I meant to say relatively small batches, when compared to general development without pull requests. Will specify 👍

Reply:
See also https://github.com/epiverse-trace/epiverse-trace.github.io/pull/335/commits/58f9da97792f29f9d5b83235c33248c2c26e1aef